src/intersections/rayline.jl

```diff
@@ -10,22 +10,26 @@ The intersection type can be one of four types:
 3. overlap of line and ray (OverlappingRayLine -> Ray)
 4. do not overlap nor intersect (NoIntersection)
 =#
-function intersection(f, r1::Ray{N,T}, l1::Line{N,T}) where {N,T}
-  a, b = r1(0), r1(1)
-  c, d = l1(0), l1(1)
+function intersection(f, ray::Ray{N,T}, line::Line{N,T}) where {N,T}
+  a, b = ray(0), ray(1)
+  c, d = line(0), line(1)
 
-  λ₁, λ₂, r, rₐ = intersectparameters(a, b, c, d)
+  # use a normalized point for ray parameter λ₁
+  len1 = norm(b - a)
+  b₀ = a + 1/len1 * (b - a)
+
+  λ₁, λ₂, r, rₐ = intersectparameters(a, b₀, c, d)
```

Comment:
I am really trying to understand what is happening here. I can say that this asymmetry where a new variable b0 is defined and d0 is not defined doesn't feel natural or mathematically pleasing. Can you help the reader of the code (not just me) understand what is the rationale behind this new b0 variable? You can add a comment before this line discussing this length issues, etc.

Reply:
we only need a rescaling if an intersection object has a fixed point to check (like the origin of ray or the end points of a segment). 

Since lines do not have such critical points it's not necessary to normalize its parameter. Note, that we do not use the second parameter at all in `rayline.jl` (we could use it in line 30 and write 
`return @IT CrossingRayLine line(λ₂) f # CASE 1` instead of 
`return @IT CrossingRayLine ray(λ₁/len1) f # CASE 1`